saleor/checkout/complete_checkout.py

```diff
@@ -70,7 +71,8 @@ def _get_voucher_data_for_order(checkout_info: "CheckoutInfo") -> dict:
     if voucher.usage_limit:
         increase_voucher_usage(voucher)
     if voucher.apply_once_per_customer:
-        add_voucher_usage_by_customer(voucher, checkout_info.get_customer_email())
+        customer_email = cast(str, checkout_info.get_customer_email())
```

Comment:
What if this checkout is anonymous and we add a promo code with `voucher.apply_once_per_customer=True` to this checkout?

Reply:
You cannot add a promo code to checkout without email. You need to assign an email to checkout before. There is also email validation in `CheckoutAddPromoCode` mutation.